### PR TITLE
DR-2340 Consolidate Dataset and Snapshot tabs into Dashboard tabs

### PR DESCRIPTION
**Pull request by myessail**
tabswitching

### CODE REVIEW

File: src/components/HomeView.jsx [hunk 1]
```diff
@@ -32,24 +40,40 @@ const styles = (theme) => ({
...
+    transition: 'none',
   },
 });
 
 class HomeView extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      tabValue: '/datasets',
```
Comment (nmalfroy) [inline src/components/HomeView.jsx]: I'm not sure that we want to use component state...shouldn't we just be using the redux store.  Component state should, in my opinion, be used exceedingly rarely

File: src/routes/Private.jsx [hunk 2]
```diff
@@ -115,13 +82,10 @@ class Private extends React.Component {
                   <div className={classes.component}>
                     <Switch>
                       <Route exact path="/" component={HomeView} />
-                      {features.searchui && (
-                        <Route exact path="/explorer" component={DataExplorerView} />
-                      )}
-                      <Route exact path="/datasets" component={DatasetsView} />
+                      <Route exact path="/datasets" component={HomeView} />
+                      <Route exact path="/snapshots" component={HomeView} />
                       <Route exact path="/datasets/:uuid/query" component={QueryView} />
```
Comment (nmalfroy) [inline src/routes/Private.jsx:90]: presumably, we'll want these paths to live under `HomeView` too

File: src/components/HomeView.jsx (hunk 1 above)
Comment (nmalfroy) [reply src/components/HomeView.jsx]: Can't we get that from the router value from the global store?

File: src/components/HomeView.jsx [hunk 3]
```diff
@@ -32,59 +39,80 @@ const styles = (theme) => ({
...
-    const { dispatch } = this.props;
-    dispatch(getDatasets(5));
-    dispatch(getSnapshots(5));
-  }
+  static prefixMatcher = new RegExp('/[^/]*');
 
   render() {
-    const { classes, datasets, features, snapshots } = this.props;
+    const { classes, location } = this.props;
+    const tabValue = HomeView.prefixMatcher.exec(location.pathname)[0];
```
Comment (nmalfroy) [inline src/components/HomeView.jsx:77]: Does this make sense as part of a component? I'm also surprised this isn't something that's part of the router

**Review by nmalfroy — APPROVED** [review]
looks good to me! (modulo linting/test failures)

File: src/components/HomeView.jsx (hunk 3 above)
Comment (myessail) [reply src/components/HomeView.jsx:77]: I think if we want the tabs in the HomeView then we would need to handle the values here in some way, this is getting the location from the global state but I could try passing it through as a prop from the router if you think that might be a better approach?

File: src/routes/Private.jsx (hunk 2 above)
Comment (myessail) [reply src/routes/Private.jsx:90]: I don't think we need to since these pages aren't under the datasets/snapshots tab in the new mocks

**Review by snf2ye — APPROVED** [review]

**Pull request merged**